plugins/netskope/icon_netskope/util/api.py

```diff
@@ -0,0 +1,137 @@
+import base64
+import time
+from logging import Logger
+from typing import Callable, Optional, Union
+
+import requests
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+RETRY_MESSAGE = "Rate limiting error occurred. Retrying in {delay:.1f} seconds ({attempts_counter}/{max_tries})"
+
+MAX_TRIES = 10
+
+
+class ApiClient:
+    def __init__(self, tenant: str, api_key_v1: str, api_key_v2: str, logger: Optional[Logger] = None) -> None:
+        self.api_url_v1 = f"https://{tenant}.goskope.com/api/v1/"
+        self.api_url_v2 = f"https://{tenant}.goskope.com/api/v2/"
+        self.api_key_v1 = api_key_v1
+        self.api_key_v2 = api_key_v2
+        self.test_connection = False
+        self.logger = logger
+
+    def update_file_hash_list(self, data: dict) -> dict:
+        update_file_hash_list_url = f"{self.api_url_v1}updateFileHashList/"
+        return self._call_api_v1("GET", update_file_hash_list_url, json_data={"token": self.api_key_v1, **data})
+
+    def get_all_url_list(self, params: dict) -> list:
+        get_all_url_list_url = f"{self.api_url_v2}policy/urllist/"
+        return self._call_api_v2("GET", get_all_url_list_url, params=params)
+
+    def create_a_new_url_list(self, params: dict) -> list:
+        create_a_new_url_list_url = f"{self.api_url_v2}policy/urllist/"
+        created_urllist = self._call_api_v2("POST", create_a_new_url_list_url, json_data=params)
+        return self.get_url_list_by_id(created_urllist[0].get("id"))
```

Comment:
Are we sure that `created_urllist` always has a first element?

Reply:
According to the API it always creates single element in database that contains list of URLs. The way of getting first element of a list as [0] is because that API return list in its output.